Panic when selecting on empty search result

Reproduce:
`gitin log`
`/`
`type in some gibberish returning no results`
`<enter>`
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/isacikgoz/gitin/prompt.(*Log).onSelect(0xc0000d4500, 0xc0000e3d50)
	/Users/ibrahim/git/gitin/prompt/log.go:62 +0x3be
github.com/isacikgoz/gitin/prompt.(*prompt).assignKey(0xc0000a8140, 0xc00000000a, 0x2)
	/Users/ibrahim/git/gitin/prompt/prompt.go:203 +0xee
github.com/isacikgoz/gitin/prompt.(*prompt).innerRun(0xc0000a8140, 0x4432ab0, 0xc0000a8140)
	/Users/ibrahim/git/gitin/prompt/prompt.go:143 +0x19f
github.com/isacikgoz/gitin/prompt.(*prompt).start(0xc0000a8140, 0x0, 0x0)
	/Users/ibrahim/git/gitin/prompt/prompt.go:111 +0x241
github.com/isacikgoz/gitin/prompt.(*Log).Start(0xc0000d4500, 0xc0000d8cc0, 0x4399800, 0xc0000f0190)
	/Users/ibrahim/git/gitin/prompt/log.go:55 +0x3fd
main.run(0x442272b, 0x3, 0xc000016104, 0x32, 0x4, 0xc0000ea3f0)
	/Users/ibrahim/git/gitin/main.go:71 +0x29e
main.main()
	/Users/ibrahim/git/gitin/main.go:24 +0x69
```

I think this is unsafe accessing here:  https://github.com/isacikgoz/gitin/blob/master/prompt/log.go#L62

Awesome tool! This is pretty low priority I think but would be cool to handle it nicer!

--- a/main.go
+++ b/main.go
@@ -33,7 +33,7 @@ func evalArgs() string {
 	pin.Command("status", "Show working-tree status. Also stage and commit changes.")
 	pin.Command("branch", "Show list of branches.")
 
-	pin.Version("gitin version 0.2.0")
+	pin.Version("gitin version 0.2.1")
 
 	pin.UsageTemplate(pin.DefaultUsageTemplate + additionalHelp() + "\n")
 	pin.CommandLine.HelpFlag.Short('h')

--- a/prompt/branch.go
+++ b/prompt/branch.go
@@ -1,6 +1,7 @@
 package prompt
 
 import (
+	"fmt"
 	"os/exec"
 
 	"github.com/fatih/color"
@@ -39,7 +40,6 @@ func (b *Branch) Start(opts *Options) error {
 	b.prompt = &prompt{
 		list:      list,
 		opts:      opts,
-		layout:    branch,
 		selection: b.onSelect,
 		keys:      b.onKey,
 		info:      b.branchInfo,
@@ -51,6 +51,9 @@ func (b *Branch) Start(opts *Options) error {
 
 func (b *Branch) onSelect() bool {
 	items, idx := b.prompt.list.Items()
+	if idx == NotFound {
+		return false
+	}
 	branch := items[idx].(*git.Branch)
 	args := []string{"checkout", branch.Name}
 	cmd := exec.Command("git", args...)
@@ -92,6 +95,9 @@ func (b *Branch) branchInfo(item Item) [][]term.Cell {
 
 func (b *Branch) deleteBranch(mode string) error {
 	items, idx := b.prompt.list.Items()
+	if idx == NotFound {
+		return fmt.Errorf("there is no item to delete")
+	}
 	branch := items[idx].(*git.Branch)
 	cmd := exec.Command("git", "branch", "-"+mode, branch.Name)
 	cmd.Dir = b.Repo.Path()

--- a/prompt/log.go
+++ b/prompt/log.go
@@ -1,6 +1,7 @@
 package prompt
 
 import (
+	"fmt"
 	"strconv"
 	"strings"
 
@@ -45,7 +46,6 @@ func (l *Log) Start(opts *Options) error {
 	l.prompt = &prompt{
 		list:      list,
 		opts:      opts,
-		layout:    log,
 		keys:      l.onKey,
 		selection: l.onSelect,
 		info:      l.logInfo,
@@ -59,6 +59,9 @@ func (l *Log) Start(opts *Options) error {
 func (l *Log) onSelect() bool {
 	// s.showDiff()
 	items, idx := l.prompt.list.Items()
+	if idx == NotFound {
+		return false
+	}
 	item := items[idx]
 	switch item.(type) {
 	case *git.Commit:
@@ -94,7 +97,10 @@ func (l *Log) onSelect() bool {
 
 func (l *Log) onKey(key rune) bool {
 	items, idx := l.prompt.list.Items()
-	item := items[idx]
+	var item Item
+	if idx != NotFound {
+		item = items[idx]
+	}
 	switch item.(type) {
 	case *git.Commit:
 		switch key {
@@ -120,6 +126,9 @@ func (l *Log) onKey(key rune) bool {
 
 func (l *Log) showDiff() error {
 	items, idx := l.prompt.list.Items()
+	if idx == NotFound {
+		return fmt.Errorf("there is no item to show diff")
+	}
 	commit := items[idx].(*git.Commit)
 	args := []string{"show", commit.Hash}
 	return popGitCommand(l.Repo, args)
@@ -127,6 +136,9 @@ func (l *Log) showDiff() error {
 
 func (l *Log) showStat() error {
 	items, idx := l.prompt.list.Items()
+	if idx == NotFound {
+		return fmt.Errorf("there is no item to show diff")
+	}
 	commit := items[idx].(*git.Commit)
 	args := []string{"show", "--stat", commit.Hash}
 	return popGitCommand(l.Repo, args)

--- a/prompt/prompt.go
+++ b/prompt/prompt.go
@@ -12,16 +12,6 @@ import (
 	"github.com/isacikgoz/gitin/term"
 )
 
-type promptType int
-
-const (
-	status promptType = iota
-	log
-	file
-	branch
-	stash
-)
-
 type keyEvent struct {
 	ch  rune
 	err error
@@ -48,8 +38,6 @@ type promptState struct {
 }
 
 type prompt struct {
-	layout promptType
-
 	list      *List
 	keys      onKey
 	selection onSelect
@@ -112,12 +100,11 @@ func (p *prompt) start() error {
 		return err
 	}
 
-	// if p.exitMsg != nil {
 	for _, cells := range p.exitMsg {
 		p.writer.WriteCells(cells)
 	}
 	p.writer.Flush()
-	// }
+
 	return nil
 }
 
@@ -160,13 +147,15 @@ mainloop:
 func (p *prompt) render() {
 	// lock screen mutex
 	p.mx.Lock()
-	defer p.mx.Unlock()
+	defer func() {
+		p.writer.Flush()
+		p.mx.Unlock()
+	}()
 
 	if p.helpMode {
 		for _, line := range genHelp(p.allControls()) {
 			p.writer.WriteCells(line)
 		}
-		p.writer.Flush()
 		return
 	}
 
@@ -188,9 +177,6 @@ func (p *prompt) render() {
 	} else {
 		p.writer.WriteCells(term.Cprint("Not found.", color.FgRed))
 	}
-
-	// finally, discharge to terminal
-	p.writer.Flush()
 }
 
 func (p *prompt) assignKey(key rune) bool {

--- a/prompt/status.go
+++ b/prompt/status.go
@@ -47,7 +47,6 @@ func (s *Status) Start(opts *Options) error {
 	s.prompt = &prompt{
 		list:      l,
 		opts:      opts,
-		layout:    status,
 		keys:      s.onKey,
 		selection: s.onSelect,
 		info:      s.branchInfo,
@@ -70,7 +69,6 @@ func (s *Status) onSelect() bool {
 
 func (s *Status) onKey(key rune) bool {
 	var reqReload bool
-
 	switch key {
 	case ' ':
 		reqReload = true
@@ -178,6 +176,9 @@ func (s *Status) hunkStage() error {
 // pop git diff
 func (s *Status) showDiff() error {
 	items, idx := s.prompt.list.Items()
+	if idx == NotFound {
+		return fmt.Errorf("there is no item to show diff")
+	}
 	entry := items[idx].(*git.StatusEntry)
 	return popGitCommand(s.Repo, fileStatArgs(entry))
 }

